### ISSUE TITLE
Can't I train KG with 30xx graphics cards?

### ISSUE THREAD

**Issue by 22nsuk** [lightvector/KataGo#367]
Hi. I want to train KG with 30xx graphics card.
But the training doesn't work with 30xx graphics cards.
I think it's because tensorflow 1.15 doesn't support cuda11 and cudn8, is that right?
So I tried to update your Python on Tensorflow 2.4, but it also failed.
Can you check if it's my personal problem or if it's normal to not be implemented?
If it is normal that it is not implemented, can you solve this part?
Please give me an answer. Thank you.

**Comment by lightvector** [comment]
Right, KataGo's training scripts require tensorflow 1.15 right now. So you can only do whatever is compatible with tensorflow 1.15, on the training side. (The selfplay and user engine side is much more up-to-date and flexible, of course).

Upgrading the training architecture and pipeline to use a newer version of Tensorflow 2.x or perhaps to use Pytorch would be fantastic. It would be a lot of work, but probably is worth doing. If you're willing to help with that work, let me know. It would probably *not* be a quick/small project, as you probably found out trying to upgrade to Tensorflow 2.4, and of course doing it to reasonable completion would be important - for this kind of thing something only halfway with missing components is not much more useful than nothing.

As far as translation to Pytorch, see https://discord.com/channels/417022162348802048/583775968804732928/776813675650220093 - which was started by @isty2e and looks like an amazing work/foundation so far. This is something I have on my todo list to try to integrate at some point, but haven't gotten around to.

**Comment by zotmeister** [comment]
Will this problem prevent contributions to distributed KG from owners of newer cards?

**Comment by Friday9i** [comment]
@zotmeister  It only impacts training, not selfplay : hence, no issue for RTX30x0 owners participating to KG Distributed :-)

**Comment by iopq** [comment]
@22nsuk 
check out my fork https://github.com/iopq/KataGo/tree/tf2

it runs on tf2, but I only tested by running it, I don't know if I introduced any bugs
